src/libsync/creds/oauth.cpp

```diff
@@ -466,14 +466,12 @@ QUrl OAuth::authorisationLink() const
 
     const QByteArray code_challenge = QCryptographicHash::hash(_pkceCodeVerifier, QCryptographicHash::Sha256)
                                           .toBase64(QByteArray::Base64UrlEncoding | QByteArray::OmitTrailingEquals);
-    QUrlQuery query { { QStringLiteral("response_type"), QStringLiteral("code") },
-        { QStringLiteral("client_id"), _clientId },
-        { QStringLiteral("redirect_uri"), QStringLiteral("%1:%2").arg(_redirectUrl, QString::number(_server.serverPort())) },
-        { QStringLiteral("code_challenge"), QString::fromLatin1(code_challenge) },
-        { QStringLiteral("code_challenge_method"), QStringLiteral("S256") },
-        { QStringLiteral("scope"), Theme::instance()->openIdConnectScopes() },
-        { QStringLiteral("prompt"), Theme::instance()->openIdConnectPrompt() },
-        { QStringLiteral("state"), QString::fromUtf8(_state) } };
+    QUrlQuery query{{QStringLiteral("response_type"), QStringLiteral("code")}, {QStringLiteral("client_id"), _clientId},
+        {QStringLiteral("redirect_uri"), QStringLiteral("%1:%2").arg(_redirectUrl, QString::number(_server.serverPort()))},
+        {QStringLiteral("code_challenge"), QString::fromLatin1(code_challenge)}, {QStringLiteral("code_challenge_method"), QStringLiteral("S256")},
+        {QStringLiteral("scope"), QString::fromUtf8(QUrl::toPercentEncoding(Theme::instance()->openIdConnectScopes()))},
```

Comment:
Can't we URL encode the entire result? Or does that cause any trouble?

Reply:
https://doc.qt.io/qt-6/qurl.html#toEncoded

Reply:
That was my idea as well. I thought we could just build the URL as-is and return `url.toEncoded()`, basically. Encoding selected parameters doesn't ensure that the end result will work, as more components might require encoding. Also, I could imagine that already encoded parameters might be re-encoded (i.e., broken) by `toEncoded()` later on.

Reply:
A couple lines below is:
```
        const QString davUser = QString::fromUtf8(QUrl::toPercentEncoding(_davUser)); // Issue #7762;
        // open id connect
        query.addQueryItem(QStringLiteral("login_hint"), davUser);
```
So I copied that. Also, we need to return a `QUrl`, and the encoding gives a `QByteArray`. Also, `QDesktopServices::openUrl` takes a URL, so my guess is that the parameters should be properly encoded, and that the url is passed without extra encoding to the system webbrowser.

Reply:
Quoting `addQueryItem`'s docs, "the key and value strings are expected to be in percent-encoded form", so we should pipe _all_ values through `toPercentEncoding` anyway. Nevertheless, we should use `QUrl::toEncoded` in the end, as the parameters might be decoded otherwise.

I think we need unit tests for this....

Reply:
done